File: benchmarks/Java/src/cd/Collision.java
package cd;


public final class Collision {
  public final CallSign aircraftA;
  public final CallSign aircraftB;
  public final Vector3D position;

  public Collision(final CallSign aircraftA, final CallSign aircraftB,
      final Vector3D position) {
    this.aircraftA = aircraftA;
    this.aircraftB = aircraftB;
    this.position = position;
  }
}

File: benchmarks/C++/src/CD/Collision.cpp
#include "Collision.h"

using namespace std;

namespace CD {
Collision::Collision(shared_ptr<CallSign> aircraftA,
                     shared_ptr<CallSign> aircraftB,
                     shared_ptr<Vector3D> position) {
  _aircraftA = aircraftA;
  _aircraftB = aircraftB;
  _position = position;
}
}  // namespace CD